.pre-commit-config.yaml

```diff
@@ -209,21 +209,24 @@ repos:
     - tests/
     pass_filenames: false
 
-- repo: local
+- repo: https://github.com/pre-commit/mirrors-pylint
+  rev: v3.0.0a4
   hooks:
   - id: pylint
-    language: system
-    name: PyLint
-    files: \.py$
-    entry: python -m pylint
     args:
     # FIXME: uncomment once it doesn't have violations
     # - docs/
     - share/
     - src/
     - tests/
+    additional_dependencies:
+      - ansible-core
```

Comment:
This introduces differing indentation, keep it the same as above.
Also, what is the justification for maintaining a copy-paste of the dependency list and loosing the ability to pin the transitive deps via constraints/lock files?

Reply:
Reopening since this was marked resolved without any reason or explanation.

Reply:
these appear to be indented like other entries in the file:
```
    additional_dependencies:
    - ansible-core
    # astroid is a transitive dep for pylint, > 2.9.0 introduces false no-name-in-module and no-member errors
    # attempt to remove this when astroid 2.9.4 or later is released
    # related: https://github.com/PyCQA/pylint/issues/5131
    # related: https://github.com/ansible/ansible-navigator/issues/772
    - astroid == 2.9.0
    - ansible-runner
    - jinja2
    - libtmux
    - onigurumacffi
    - pylint-pytest < 1.1.0
    - setuptools-scm
    - sphinx
```

Reply:
I know. Although, the second part of the comment was never addressed.

Reply:
researching, so I grok it

Reply:
Is this because we are using the native pre-commit hook rather than local which would use the `test-requirements` file?

Reply:
That's exactly my concern — the inability to pin the whole dependency tree to avoid transitive dependencies causing problems, deeper than the direct one (like `astroid`). Why is it better to lose this ability?

Reply:
By using the native pre-commit hook, and specifying a version there, does it isolate use from the python version on the runner and therefore make it consistent until the precommit native hook version is changed?  I did see some inconsistencies with #797 (running locally vs zuul) I suspected were because of this.

Reply:
This is a good read, explaining why pre-commit will never use the requirements file, here for future ref: https://github.com/pre-commit/pre-commit/issues/730

Reply:
I should note, the list of dependencies for navigator should remain small, and new deps rarely added and only after a lot of conversation and consideration.  While this doesn't avoid the version issue, the list should remain small.

Reply:
Should we consider making a note here and in the test-requirements/tox file that they need to remain in sync?

Reply:
Feel free to resolve this once the last comment in the main thread is addressed.

Reply:
> @ssbarnea I'll add a better description with some pros/cons and rollback details in the AM if you don't beat me to it.

Feel free to alter it and merge it. As long my `alias pc=pre-commit run -a` includes execution of pylint, we are all ok.

As a note, we already had this setup in all the other repos and it was fairly stable. From linting perspective version of dependencies is far less important than testing code. I am in favour of pinning for unit and worked with Sviat to do it.

I already think we spent too much time on simple change like this, so I will let you deal the final blow ;)

> @ssbarnea I'll add a better description with some pros/cons and rollback details in the AM if you don't beat me to it.

Feel free to alter it and merge it. As long my `alias pc=pre-commit run -a` includes execution of pylint, we are all ok.

As a note, we already had this setup in all the other repos and it was fairly stable. From linting perspective version of dependencies is far less important than testing code. I am in favour of pinning for unit and worked with Sviat to do it.

I already think we spent too much time/effort on simple change like this, so I will @cidrblock polish it and merge it so we avoid increasing the thread on this change.